src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -121,6 +201,28 @@ export class ResultComponent implements OnInit, OnChanges {
         return this.result!.hasFeedback;
     }
 
+    hasParticipationResults(): boolean {
+        if (!this.participation) {
+            return false;
+        }
+        return this.participation.results && this.participation.results.length > 0;
+    }
+
+    isSubmissionInDueTime(): boolean {
+        if (this.participation.submissions[0] && this.participation.submissions[0].submissionDate && this.participation.exercise.dueDate) {
+            this.participation.submissions[0].submissionDate = moment(this.participation.submissions[0].submissionDate);
+            return this.participation.submissions[0].submissionDate.isBefore(this.participation.exercise.dueDate);
+        } else if (!this.participation.exercise.dueDate) {
```

Comment:
In my opinion, this can be simplified to `else { this.participation.exercise.dueDate ? false : true }`

Reply:
You're totally right but I guess `return !this.participation.exercise.dueDate` is even easier :-)